### ISSUE TITLE
Operator pod is missing multus nework annotation with external -> enabled: true

### ISSUE THREAD

**Issue by mazocode** [rook/rook#7563]
* Bug Report

**Deviation from expected behavior:**
The operator is using the pod network to contact the external ceph cluster instead of the specified multus network.

**Expected behavior:**
Operator deployment template including the multus network annotation.

**How to reproduce it (minimal and precise):**
Try to use rook to access an external cluster trough a multus network. Example configuration:
```
cat <<EOY | kubectl --context=isc apply -f -
apiVersion: ceph.rook.io/v1
kind: CephCluster
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  external:
    enable: true
  network:
    provider: multus
    selectors:
      public: default/my-storage-network
  crashCollector:
    disable: true
  dataDirHostPath: /var/lib/rook
  cephVersion:
    image: ceph/ceph:v15.2.9
EOY
```

This has an indirect connection with https://github.com/rook/rook/issues/7550 together (ceph-mgr cannot be exposed for external access).

**Environment**:
Rook version (use rook version inside of a Rook Pod): 1.5.9
Storage backend version (e.g. for ceph do ceph -v): 15.2.9
Kubernetes version (use kubectl version): v1.20.2
Storage backend status (e.g. for Ceph use ceph health): HEALTH_OK

**Comment by rohan47** [comment]
@mazocode Currently, we are not applying the public network to the rook-ceph-operator pod on purpose as we know that the pod network is sufficient for the rook-ceph-operator pod. Also can you please point out a use-case where it is a must that rook-ceph-operator uses the public network? 

It will be good to have the operator pod use multus annotations. 
It can be done in two ways:
1. Apply the public network annotation manually to the operator deployment.
2. The rook-ceph-operator gets the multus settings from the CephCluster spec, after completing reconciliation the operator applies the annotations to the rook-ceph-operator deployment which restarts the operator.

**Comment by mazocode** [comment]
> Also can you please point out a use-case where it is a must that rook-ceph-operator uses the public network?

I am trying to expose a rook-ceph cluster for acces from different clusters on the network. They all can talk trough a dedicated storage network (attached using multus) and trough the cluster network. The POD network does not get routed outside a cluster. 

There are several issues we've encountered when trying to expose the ceph services for access from outside the cluster. This particular issue refers to the operator trying to contact the ceph services at an address from within a dedicated storage network. Without the storage network attached to the operator pod, it will travel trough the regular network paths which do not grant access to the dedicated storage network. Even if we would allow this kind of communication (which introduces additional overhead for management on the firewalls), it would travel an inefficient path. While this might not be an issue with the operator, it kind of defeats the purpose of the dedicated network.

**Comment by rohan47** [comment]
> > Also can you please point out a use-case where it is a must that rook-ceph-operator uses the public network?
> 
> I am trying to expose a rook-ceph cluster for acces from different clusters on the network. They all can talk trough a dedicated storage network (attached using multus) and trough the cluster network. The POD network does not get routed outside a cluster.

Have you tried appling the annotations manually to the rook-ceph-operator pod in the `rook-ceph-operator` deployment to confirm applying public network to the rook-ceph-operator pod works as a workaround

Something like:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: rook-ceph-operator
  namespace: rook-ceph # namespace:operator
  labels:
    operator: rook
    storage-backend: ceph
spec:
  selector:
    matchLabels:
      app: rook-ceph-operator
  replicas: 1
  template:
    metadata:
      labels:
        app: rook-ceph-operator
      annotations:
        k8s.v1.cni.cncf.io/networks: public
    spec:
.
.
.
````
> 
> There are several issues we've encountered when trying to expose the ceph services for access from outside the cluster. This particular issue refers to the operator trying to contact the ceph services at an address from within a dedicated storage network. Without the storage network attached to the operator pod, it will travel trough the regular network paths which do not grant access to the dedicated storage network. Even if we would allow this kind of communication (which introduces additional overhead for management on the firewalls), it would travel an inefficient path. While this might not be an issue with the operator, it kind of defeats the purpose of the dedicated network.

Okay, so we need a fix for this.

**Comment by travisn** [comment]
@rohan47 Do we just need to document adding this annotation? Or might the issue be fixed now with #8272?

**Comment by rohan47** [comment]
> @rohan47 Do we just need to document adding this annotation? Or might the issue be fixed now with #8272?

The issue is fixed by #8272.
We don't need annotations on the operator pod.

**Comment by leseb** [comment]
> > @rohan47 Do we just need to document adding this annotation? Or might the issue be fixed now with #8272?
> 
> The issue is fixed by #8272.
> We don't need annotations on the operator pod.

It's not fixed by #8272 since in external mode the mgr pod is not running. Rook will likely never set the annotations on the operator pod, so it's still nice to document this.
